lib/utils/combineJSON.js

```diff
@@ -99,19 +99,40 @@ export default async function combineJSON(
 
       // If there is no file_content then no custom parser ran on that file
       if (!file_content) {
-        if (['.js', '.mjs'].includes(extname(filePath))) {
-          let resolvedPath = resolve(filePath, vol?.__custom_fs__);
-          // eslint-disable-next-line no-undef
-          if (isNode && process?.platform === 'win32') {
-            // Windows FS compatibility. If in browser, we use an FS shim which doesn't require this Windows workaround
-            resolvedPath = new URL(`file:///${resolvedPath}`).href;
+        switch (extname(filePath)) {
+          case '.js':
+          case '.mjs':
+          case '.ts': {
+            let resolvedPath = resolve(filePath, vol?.__custom_fs__);
+            // eslint-disable-next-line no-undef
+            if (isNode && process?.platform === 'win32') {
+              // Windows FS compatibility. If in browser, we use an FS shim which doesn't require this Windows workaround
+              resolvedPath = new URL(`file:///${resolvedPath}`).href;
+            }
+            file_content = (await import(/* @vite-ignore */ /* webpackIgnore: true */ resolvedPath))
+              .default;
+            break;
+          }
+          case '.json':
+          case '.jsonc':
+          case '.json5': {
+            file_content = JSON5.parse(
+              /** @type {string} */ (volume.readFileSync(resolvedPath, 'utf-8')),
+            );
+            break;
+          }
+          default: {
+            // Use json parser fallback by default
+            // Other file types like .hjson, .topojson, or .ndjson, might be handled with dedicated parsers in the future.
+            // This warning is here to hint users that their file is parsed with the standard JSON5 parser and not any dedicated parser.
+            // Users should open an issue and provide the use case that caused the issue so we can investigate if we should allow users to inject custom parsers or add dedicated support.
+            console.warn(
+              `Unrecognized file extension: ${extname(filePath)}. Using JSON5 parser. If you see this open an issue on the Style Dictionary GitHub: https://github.com/amzn/style-dictionary/issues`,
+            );
```

Comment:
```suggestion
            console.warn(
              `Unrecognized file extension: ${extname(filePath)}. Using JSON5 parser as a default. Alternatively, create a custom parser to handle this filetype https://styledictionary.com/reference/hooks/parsers/`,
            );
```
I don't think it makes sense for them to create an issue unless there is a specific reason why they feel like Style Dictionary should add default support for it. Majority of cases I think will be that creating a custom parser is the way to go.